records/state/state.go

```diff
@@ -103,7 +116,18 @@ func (t *Task) containerIP(src string) string {
 			continue
 		}
 
-		// find the latest docker-inspect label
+		// first try to extract the address from the NetworkInfo structure
+		if len(status.ContainerStatus.NetworkInfos) > 0 {
+			// TODO(CD): handle multiple NetworkInfo objects
+			// TODO(CD): only create A records if the address is IPv4
+			// TODO(CD): create AAAA records if the address is IPv6
+			latestContainerIP = status.ContainerStatus.NetworkInfos[0].IPAddress
+			latestTimestamp = status.Timestamp
+			break
```

Comment:
You're right, I misread the following break statement.  I did wonder whether we were assuming sorted updates, and if so why bother to check the timestamp.  That makes more sense.  This should be `continue` instead to jump over the label inspection (to maintain preference for NetworkInfo addresses), but still consider the other elements.  Good catch.